scarb/src/core/registry/mod.rs

```diff
@@ -11,7 +13,15 @@ pub mod patch_map;
 pub mod patcher;
 pub mod source_map;
 
-pub const DEFAULT_REGISTRY_INDEX: &str = "https://there-is-no-default-registry-yet.com";
+static DEFAULT_REGISTRY_INDEX: Lazy<&'static str> = Lazy::new(|| {
+    let registry_url = env::var("SCARB_REGISTRY_URL")
+        .unwrap_or_else(|_| "https://there-is-no-default-registry-yet.com".into());
+    Box::leak(registry_url.into_boxed_str())
```

Comment:
I would be happy to hear any thoughts about this, as I'm not sure about its implications. Box::leak is used to convert the String to a &'static str by leaking the heap-allocated memory. This should be safe in this context because Lazy ensures the value is only initialized once and lives for the entire program duration, but still this may not be THE optimal way to achieve this.

Reply:
You can technically keep `Lazy<String>` there and it'd be more collect.

But I think this is not the best approach for implementing this. I'd rather put this into `Config`, so that this could eventually be stored in some kind of a file. After all, someone might want to use a custom proxy for example and they'd like to write the URL once to some file and leave it.

Reply:
Sure, could make more sense, thanks for pointers. 
btw is this something that, in your opinion, could be settable from the manifest file? Just wondering how I should approach this

Reply:
Definitely not from the manifest because this is something user-specific, not project

Cargo has this: https://doc.rust-lang.org/cargo/reference/config.html

Reply:
Ok cool! so based on this, and after discussing things with @maciektr , let's plan the work accordingly:

- I will first redo this PR, to read the url from a flag and hold it in the `Config` struct
- I will then implement reading this from said file, at some later date, in a different PR under this issue https://github.com/software-mansion/scarb/issues/1441

Sounds good?

Reply:
sounds good